cmd/cosign/cli/sign/sign.go

```diff
@@ -138,7 +138,26 @@ func SignCmd(ro *options.RootOptions, ko options.KeyOpts, signOpts options.SignO
 	ctx, cancel := context.WithTimeout(context.Background(), ro.Timeout)
 	defer cancel()
 
-	sv, err := SignerFromKeyOpts(ctx, signOpts.Cert, signOpts.CertChain, ko)
+	// Always default to ECDSA_SHA2_256_NISTP256 for now
+	if ko.SigningAlgorithm == pb_go_v1.KnownSignatureAlgorithm_KNOWN_SIGNATURE_ALGORITHM_UNSPECIFIED {
+		ko.SigningAlgorithm = pb_go_v1.KnownSignatureAlgorithm_ECDSA_SHA2_256_NISTP256
+	}
+
+	if !cosign.ClientAlgorithmsRegistry.HasAlgorithmDetails(ko.SigningAlgorithm) {
+		return fmt.Errorf("unsupported signing algorithm: %s", ko.SigningAlgorithm.String())
+	}
+
+	algorithmDetails, err := cosign.ClientAlgorithmsRegistry.GetAlgorithmDetails(ko.SigningAlgorithm)
+	if err != nil {
+		return err
+	}
```

Comment:
This does not make too much sense when the user specifies `--key`. What should be the behaviour in that case? Right now a user can even specifies a e.g. RSA key with `--key` and this check would not detect it (because the signing algorithm defaults to `ECDSA_SHA2_256_NISTP256`).

How do we want cosign to behave? Shall the registry block also --key algorithms? If so, we should probably add RSA to the list of accepted client signing algorithms as that is already valid in the current state.